app/battle.py

```diff
@@ -0,0 +1,31 @@
+from typing import List
+from app.knights import Knight
+
+
+class BattleKnight:
+    def __init__(self, knights: List[Knight]):
+        self.knights = knights
+
+    def battle_preparation(self) -> None:
+        for knight in self.knights:
+            knight.protection = sum(armour["protection"] for armour in knight.armour)
+            knight.power += knight.weapon["power"]
```

Comment:
Make sure that the `weapon` attribute in the `Knight` class is a dictionary with a `power` key, as assumed here. This matches the configuration, but verify the `Knight` class definition to ensure consistency.